### DISCUSSION TITLE
What purpose of the signal "Activate_OUT" on doors with buttons? Where can I read about this?

### DISCUSSION THREAD

**Discussion by pavel4win** [FakeFishGames/Barotrauma#13853] [Ideas and suggestions]
What purpose of the signal "Activate_OUT" on doors with buttons? Where can I read about this?
Is this an input or output signal?
Does this work the same as a button?
Or similar to a lever?
What signal should be sent there 1 or 0 or any non zero?
Where can I read about this?

**Comment by Regalis11** [comment]
https://github.com/FakeFishGames/Barotrauma/pull/6194

> What this PR does is adds an activate_out wiring output to doors with buttons, windowed doors with buttons, and hatches with buttons. It also allows users to suppress the default door open/close action that normally accompanies these integrated button/door assemblies. The idea is that this allows doors and hatches with integrated buttons to act as triggers for downstream logic circuits which may enable more advanced opening and closing functionality. This also has the added benefit that when AI goes to activate doors and hatches with buttons via waypoints to traverse thresholds it also triggers these custom logic circuits and #JustWorks.

Looks like it hasn't been added to the wiki yet, I think we need to update that page!

**Comment by NotWendy** [comment]
In short, it is an output signal. Sends 1 when the door is interacted with, same way a button would (pulse signal).

**Comment by pavel4win** [comment]
The two previous comments contradict each other:
Regalis11
writes "Activate_OUT" is output.
BUT if it can change the functionality of the buttons, which means this is an input ?

NotWendy
writes that "Activate_OUT" sends 1 at the moment when someone presses the door button, the rest of the time it does not send anything. I understood his explanation exactly like this. Am I right?

I tried to connect a text screen to "Activate_OUT" and didn't see any value.
I connected the button to "Activate_OUT", nothing changes when pressed.
I hope the wiki will be updated soon, to clarify this issue.

**Reply by NotWendy** [reply]
You're right in your interpretation of my answer. Try connecting activate_out to a light's toggle_state. Any time you interact with the door, the light will change its state.
Not sure about its interaction with a text display.

**Reply by pavel4win** [reply]
Thank you for your reply. So this is a way to find out whether the button on the door was pressed without dancing with a tambourine around the "state_out" signal. I will use it in my schemes. Now I can build alternative logic around "button pressing" by sending this signal to the "toggle_state" or "set_state" after logical processing. Now I understand what Regalis11 was talking about.

**Reply by Regalis11** [reply]
Yeah, the explanation I copied from the PR might not have been the simplest way to put it, thanks @NotWendy for clarifying! :) 

Btw, as a side note, you can tell whether a connection is an input or an output simply by looking at which side of the connection panel it's on. Everything at the left is an input, everything at right an output.
